dem.cc

```diff
@@ -0,0 +1,234 @@
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+// grain structure with the following members:radius, mass density, x of center,
+// y of center, translational velocity and acceleration in 2 directions,
+// rotation angle, rotational velocity, rotational acceleration, mass, moment of
+// inertia, and gravity
+struct grain {
+  double r, rho, xc, yc, vx, vy, ax, ay, theta, omega, alpha;
+  double m = rho * M_PI * r * r;
+  double inertia = 0.5 * m * r * r;
+  double gx, gy;
+  bool fix;
+};
+
+const int ngrains = 2;
+struct grain grains[ngrains];
+
+struct contact {
+  double kn;  // normal stiffness
+  double kt;  // tangent stiffness
+  // double damp_ratio;    //damping ratio
+  double cor;       // coefficient of restitution
+  double fric;      // friction coefficient (mu)
+  double rollfric;  // rolling friiction coefficient (mu_r)
+  double gamma =
+      -log(cor) /
+      sqrt(M_PI * M_PI + log(cor) * log(cor));  // needed for calculating cn
+                                                // based on cor (Cleary,2000)
+};
+
+struct contact grain_grain;
+double dt = 1e-6;  // timestep
+
+// function to calculate the forces between particles jm and js and update the
+// acceleration of jm (master particle) in 2 directions.
+void update_acceleration(int jm, int js) {
+
+  grains[jm].ax = grains[jm].gx;
+  grains[jm].ay = grains[jm].gy;
+
+  const double distance = sqrt(pow(grains[jm].xc - grains[js].xc, 2) +
+                               pow(grains[jm].yc - grains[js].yc, 2));
+  const double delta = distance - grains[jm].r - grains[js].r;
+  if (delta < 0.) {
+    // calculating the normal and tangent unit vectors
+    const double normal[2];
```

Comment:
This works?! you can't make it `const` and then assign a value. Please make these arrays as just doubles

Reply:
sorry:)) I was trying to be quick so I forgot to even make sure it compiles before committing. My bad.